internal/formats/common/cyclonedxhelpers/cpe.go

```diff
@@ -1,12 +1,30 @@
 package cyclonedxhelpers
 
-import "github.com/anchore/syft/syft/pkg"
+import (
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/pkg"
+)
 
-func CPE(p pkg.Package) string {
+func encodeCPE(p pkg.Package) string {
 	// Since the CPEs in a package are sorted by specificity
 	// we can extract the first CPE as the one to output in cyclonedx
 	if len(p.CPEs) > 0 {
 		return pkg.CPEString(p.CPEs[0])
 	}
 	return ""
 }
+
+func decodeCPEs(c *cyclonedx.Component) []pkg.CPE {
+	// FIXME we not encoding all the CPEs (see above), so here we just use the single provided one
```

Comment:
This is something we've decided to do as part of this PR, this is a known deficiency that needs to be addressed at some point soon. It probably can have CPEs generated automatically or something similar, like is in the Grype PR here: https://github.com/anchore/grype/pull/606

Reply:
@kzantow this might help -> https://github.com/anchore/syft/pull/819